api/v1alpha1/dnsrecord_types.go

```diff
@@ -182,6 +101,33 @@ const (
 	DefaultGeo string = "default"
 )
 
+// GetRootDomain returns the shortest domain that is shared across all spec.Endpoints dns names.
+// Validates that all endpoints share an equal root domain and returns an error if they don't.
+func (s *DNSRecord) GetRootDomain() (string, error) {
```

Comment:
doesn't this just get us the longest shared hostname?

e.g. with hosts like:
```
app1.pb.domain.com.
app2.pb.domain.com.
app3.pb.domain.com.
```
We'd get not value? Where we'd expect `pb.domain.com` right?

Doesn't it make more sense to work up the first hostname sections until we find the longest section that every host has? I.e. in the example above, they all have `com`, `domain.com`, `pb.domain.com` with that being the longest value that every host has, return that?

Reply:
Yeah, likely not going to work if hosts are of identical length, will rework. That said i don't want to spend too much time on it as i think we will just replace it with a spec update tbh, just don't want to do that here.

Reply:
Hmm actually, the current logic is correct, or at least correct with what i was thinking. 

We are trying to find a single entry that is the start point, not work out the longest shared hostname. We are assuming that all the records in this record are related and that they all share a dns name that is also defined in this record set.

In your example we would expect there to be an entry for `pb.domain.com`, or you create 3 dns records.

I suppose it depends what we are trying to support with our DNSRecord, i was looking at it from the POV that a  kuadrant DNSRecord is defining a record set from beginning to end and if you have multiple endpoints one of them *must* be the start of the chain. :thinking:

Reply:
hmm yeah, that's true, so I suppose this functions for a DNS Record created by Kuadrant, but won't for like a weird manual DNS Record... but then in that scenario there might not even be a "root" it could just be a set of unrelated A records, right?

This is making me wonder if the "Root Host" is actually a Kuadrant concept, and not a dns-operator concept, and so doesn't belong here?

Reply:
We have a follow up issue now to add a "rootHost" to the DNSRecord spec the https://github.com/Kuadrant/dns-operator/issues/50. This will allow the enforcement of the desired behaviour here, but also allow a DNSRecord to be used for other purposes that do not require a shared "rootHost".